tests/vm_migration_test.go

```diff
@@ -0,0 +1,128 @@
+			Expect(se.ErrStatus.Code).To(Equal(int32(404)))
+			Expect(r).To(BeNil())
+
+		})
+
+		It("Should fail if the VM does not exist", func() {
+			r, err := restClient.Get().Resource("vms").Namespace(api.NamespaceDefault).Name(vm.ObjectMeta.Name).Do().Get()
+			var se *errors.StatusError = err.(*errors.StatusError)
+			Expect(se.ErrStatus.Code).To(Equal(int32(404)))
+			Expect(r).To(BeNil())
+
+			err = restClient.Post().Resource("migrations").Namespace(api.NamespaceDefault).Body(migration).Do().Error()
+			Expect(r).To(BeNil())
+			//wait for the  controller to update status
+			time.Sleep(time.Second)
+
+			r, err = restClient.Get().Resource("migrations").Namespace(api.NamespaceDefault).Name(migration.ObjectMeta.Name).Do().Get()
+			Expect(err).To(BeNil())
+
+			var m *v1.Migration = r.(*v1.Migration)
+			Expect(m.Status.Phase).To(Equal(v1.MigrationFailed))
+			restClient.Delete().Resource("migrations").Namespace(api.NamespaceDefault).Name(migration.ObjectMeta.Name).Do()
+
+		})
+
+		It("Should Succeed once the VM exists", func() {
+
+			obj, err := restClient.Post().Resource("vms").Namespace(api.NamespaceDefault).Body(vm).Do().Get()
+			Expect(err).To(BeNil())
+
+			tests.NewObjectEventWatcher(obj, func(event *kubev1.Event) bool {
+				//Expect(event.Type).NotTo(Equal("Warning"), "Received VM warning event")
+				if event.Type == "Normal" && event.Reason == v1.Started.String() {
+					obj, err := restClient.Get().Namespace(api.NamespaceDefault).
+						Resource("vms").Name(vm.GetObjectMeta().GetName()).Do().Get()
+					Expect(err).To(BeNil())
+					Expect(string(obj.(*v1.VM).Status.Phase)).To(Equal(string(v1.Running)))
+					//close(done)
+					return true
+				}
+				return false
+			}).Watch()
+
+			Expect(err).To(BeNil())
+			var r runtime.Object = nil
+
+			time.Sleep(time.Second * 3)
+
+			r, err = restClient.Get().Resource("vms").Namespace(api.NamespaceDefault).Name(vm.ObjectMeta.Name).Do().Get()
+			var vmR *v1.VM = r.(*v1.VM)
+			Expect(vmR.ObjectMeta.Name).To(Equal(vm.ObjectMeta.Name))
+
+			//wait for the  controller to update status
+			time.Sleep(time.Second * 3)
+
+			r, err = restClient.Get().Resource("migrations").Namespace(api.NamespaceDefault).Name(migration.ObjectMeta.Name).Do().Get()
+			var se *errors.StatusError = err.(*errors.StatusError)
+			Expect(se.ErrStatus.Code).To(Equal(int32(404)))
+			Expect(r).To(BeNil())
+
+			err = restClient.Post().Resource("migrations").Namespace(api.NamespaceDefault).Body(migration).Do().Error()
+			Expect(err).To(BeNil())
+			//wait for the  controller to update status
+			time.Sleep(time.Second * 3)
+
+			r, err = restClient.Get().Resource("migrations").Namespace(api.NamespaceDefault).Name(migration.ObjectMeta.Name).Do().Get()
+			Expect(err).To(BeNil())
+
+			var m *v1.Migration = r.(*v1.Migration)
+			Expect(m.Status.Phase).To(Equal(v1.MigrationInProgress))
+			restClient.Delete().Resource("migrations").Namespace(api.NamespaceDefault).Name(migration.ObjectMeta.Name).Do()
```

Comment:
Why are you deleting the migration here? The AfterEach loop should do that for you already ...